crates/chain/src/local_chain.rs

```diff
@@ -148,6 +158,39 @@ impl CheckPoint {
     pub fn iter(&self) -> CheckPointIter {
         self.clone().into_iter()
     }
+
+    /// Query for checkpoint at `height`.
+    ///
+    /// Returns `None` if checkpoint at `height` does not exist`.
+    pub fn query(&self, height: u32) -> Option<Self> {
+        self.iter()
+            // optimization to avoid iterating the entire chain if we do not get a direct hit
+            .take_while(|cp| cp.height() >= height)
+            .find(|cp| cp.height() == height)
```

Comment:
Perhaps opportunity to reuse `range`, since `query` is its special case? It seems that even performance-wise it would not be worse.

```suggestion
	self.range(height..=height).next()
```

Reply:
Changed.